test-d/camel-case.ts

```diff
@@ -0,0 +1,47 @@
+import {CamelCase} from '../source/camel-case';
+import {expectType, expectAssignable} from 'tsd';
+
+// CamelCase
+const camelFromKebab: CamelCase<'foo-bar'> = 'fooBar';
+expectType<'fooBar'>(camelFromKebab);
+
+const camelFromComplexKebab: CamelCase<'foo-bar-abc-123'> = 'fooBarAbc123';
+expectType<'fooBarAbc123'>(camelFromComplexKebab);
+
+const camelFromSpace: CamelCase<'foo bar'> = 'fooBar';
+expectType<'fooBar'>(camelFromSpace);
+
+const camelFromSnake: CamelCase<'foo_bar'> = 'fooBar';
+expectType<'fooBar'>(camelFromSnake);
+
+const noDelimiterFromMono: CamelCase<'foobar'> = 'foobar';
+expectType<'foobar'>(noDelimiterFromMono);
+
+const camelFromMixed: CamelCase<'foo-bar_abc xyzBarFoo'> = 'fooBarAbcXyzBarFoo';
+expectType<'fooBarAbcXyzBarFoo'>(camelFromMixed);
+
+const camelFromVendorPrefixedCssProperty: CamelCase<'-webkit-animation'> = 'WebkitAnimation';
+expectType<'WebkitAnimation'>(camelFromVendorPrefixedCssProperty);
```

Comment:
Lodash's `camelCase` function returns `webkitAnimation` (lowercase `w`). I don't know what other libraries we should test but if we're preferring one library over another I think we should document which behavior is picked.

If libraries all consistently return `webkitAnimation` then this should be simple enough to solve with `Uncapitalize` or some otherwise minor reworking.

Reply:
Right, I can agree with that, would require the adding of a `trim<>`, but that's doable 👍

Reply:
@knpwrs Fixed it 👍